app/test_main.py

```diff
@@ -1 +1,47 @@
-# write your code here
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "validation_result, internet_connection_result, access_to_page",
+    [
+        pytest.param(
+            False,
+            False,
+            "Not accessible",
+            id="Close access without internet connection and valid url"
+        ),
+        pytest.param(
+            False,
+            True,
+            "Not accessible",
+            id="Close access without valid url"
+        ),
+        pytest.param(
+            True,
+            False,
+            "Not accessible",
+            id="Close access without internet connection"
+        ),
+        pytest.param(
+            True,
+            True,
+            "Accessible",
+            id="Access with internet connection and valid url"
+        )
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_check_access_google_page(
+    mocked_valid_google_url: callable,
```

Comment:
wrong type annotation for mocked_valid_google_url and mocked_internet_connection